### ISSUE TITLE
Balance loss is not optimized

### ISSUE THREAD

**Issue by xptree** [laekov/fastmoe#49]
https://github.com/laekov/fastmoe/blob/295a615aacce7e54a37e7935274ba15e901c78e4/fmoe/megatron/balance.py#L97-L106

`torch.tensor(loss_list)` will remove the `grad_fn` of each loss. See https://pytorch.org/docs/stable/generated/torch.tensor.html?highlight=tensor#torch.tensor:

When data is a tensor x, torch.tensor() reads out ‘the data’ from whatever it is passed, and constructs a leaf variable. Therefore torch.tensor(x) is equivalent to x.clone().detach()

We should use `torch.cat` instead, i.e., `torch.cat([loss.unsqueeze(0) for loss in loss_list])`.
